nemo_rl/models/policy/dtensor_policy_worker.py

```diff
@@ -157,17 +162,26 @@ def __init__(
             raise ValueError(f"Unknown precision: {self.cfg['precision']}")
 
         print(f"[Rank {self.rank}] Loading model {model_name} on CPU...")
+        self.enable_seq_packing = self.cfg["sequence_packing"]["enabled"]
+        if self.enable_seq_packing:
+            print(
+                f"[Rank {self.rank}] Sequence packing is enabled for model {model_name}"
+            )
+            print(f"[Rank {self.rank}] Using FlashAttention2 for sequence packing")
         self.model = AutoModelForCausalLM.from_pretrained(
             model_name,
             device_map="cpu",  # load weights onto CPU initially
             # Always load the model in float32 to keep master weights in float32.
             # Keeping the master weights in lower precision has shown to cause issues with convergence.
             # https://github.com/NVIDIA/NeMo-RL/issues/279 will fix the issue of CPU OOM for larger models.
-            torch_dtype=torch.float32,
+            torch_dtype=self.dtype,
```

Comment:
We have seen accuracy degradation if you load the master weights with bf16 (hence the comment line in 174-176).

Reply:
Running `AutoModelForCausalLM` with flash attention was causing issues with bf16 precision.

I didn't observe any accuracy degradation with this change, but my tests were limited. If we have to load weights in FP32 then I'll think of another solution.